platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -75,24 +70,26 @@ class DefaultConfigurationCache internal constructor(
     private val cacheKey: ConfigurationCacheKey,
     private val problems: ConfigurationCacheProblems,
     private val scopeRegistryListener: ConfigurationCacheClassLoaderScopeRegistryListener,
-    private val cacheRepository: ConfigurationCacheRepository,
     private val instrumentedInputAccessListener: InstrumentedInputAccessListener,
     private val configurationTimeBarrier: ConfigurationTimeBarrier,
     private val buildActionModelRequirements: BuildActionModelRequirements,
     private val buildStateRegistry: BuildStateRegistry,
     private val virtualFileSystem: BuildLifecycleAwareVirtualFileSystem,
     private val buildOperationRunner: BuildOperationRunner,
     private val cacheFingerprintController: ConfigurationCacheFingerprintController,
-    private val resolveStateFactory: LocalComponentGraphResolveStateFactory,
     /**
      * Force the [FileSystemAccess] service to be initialized as it initializes important static state.
      */
     @Suppress("unused")
     private val fileSystemAccess: FileSystemAccess,
-    private val calculatedValueContainerFactory: CalculatedValueContainerFactory,
-    private val modelSideEffectExecutor: ConfigurationCacheBuildTreeModelSideEffectExecutor,
-    private val deferredRootBuildGradle: DeferredRootBuildGradle
-) : BuildTreeConfigurationCache, Stoppable {
+    private val deferredRootBuildGradle: DeferredRootBuildGradle,
+    host: LazyService<HostServiceProvider>,
```

Comment:
💭 I think it would be slightly nicer with the laziness expressed via the existing `Lazy<T>` type (which is basically a `Supplier`) to keep the constructors free of Gist concerns / terminology. Just like it would be better for an optional service to be expressed in terms of an existing `Optional<T>` rather than a new `OptionalService<T>` type.

Reply:
I considered this option, but the problem is that `Lazy` currently lives in `core-runtime/functional` and relies on extensively on the Java 8+ types. At the same time, Gist still has to stay Java 6 compliant.

I agree with the idea of using JDK types to express wrapping of services. Same as we use `java.util.List` for service aggregation. I think it would make sense to replace the `LazyService` type with `Supplier` (or `Lazy`), when we replace the Java 6 requirement with Java 8+ (for workers, where Gist still has to work).

In the scope of this PR, I would introduce a new type.

Reply:
Let's mention this in a comment on `LazyService` to make sure we don't have to re-discover this decision again.